tests/datasets/test_views.py

```diff
@@ -989,6 +989,27 @@ def test_dataset_members_delete_member(app: DjangoTestApp):
     assert len(mail.outbox) == 0
 
 
+@pytest.mark.haystack
+def test_dataset_management_areas(app: DjangoTestApp):
+    parent_org = OrganizationFactory()
+    child_org1 = parent_org.add_child(instance=OrganizationFactory.build(title='org-test-1'))
+    child_org2 = parent_org.add_child(instance=OrganizationFactory.build(title='org-test-2'))
+    DatasetFactory(organization=parent_org)
+    DatasetFactory(organization=child_org1)
+    DatasetFactory(organization=child_org2)
+
+    resp = app.get(reverse("dataset-list"))
+    area_facets = resp.context['management_area_facet']
+    resp = resp.click(linkid="Dataset-stats-supervisor")
+
+    max_count = 0
+    for area in area_facets:
+        if max_count < area.get('count'):
+            max_count = area.get('count')
+
+    assert resp.context['management_areas'] == area_facets
+    assert resp.context['max_count'] == max_count
```

Comment:
Please compare results with hardcoded values, because faced could also do incorrect counting. I see 3 datasets and one parent organization, so in tests, I should see single organization with 3 datasets.